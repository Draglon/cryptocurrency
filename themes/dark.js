export default {
  // React Native Elements
  Button: {
    containerStyle: {
      width: 150,
    },
    titleStyle: {
      color: '#00f',
    },
  },
  Icon: {
    type: 'font-awesome',
    size: 24,
  },

  // Custom Elements
  IconClose: {
    containerStyle: {
      backgroundColor: '#ccc',
      color: '#f00',
      position: 'absolute',
      top: 10,
      right: 10,
      width: 30,
      height: 30,
      lineHeight: 30,
      textAlign: 'center',
    },
  }
};
