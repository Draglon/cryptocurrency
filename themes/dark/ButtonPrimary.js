// import { COLORS } from '../../constants/variables';

export default {
  // Middle Button - default
  type: 'solid',
  containerStyle: {},
  buttonStyle: {
    height: 50,
  },
  titleStyle: {},
  disabledStyle: {},
  disabledTitleStyle: {},
  loadingStyle: {},
  iconContainerStyle: {},
  // Small Button
  small: {
    containerStyle: {},
    buttonStyle: {
      height: 30,
    },
    titleStyle: {},
    disabledStyle: {},
    disabledTitleStyle: {},
    loadingStyle: {},
    iconContainerStyle: {},
  },
  // Large Button
  large: {
    containerStyle: {},
    buttonStyle: {
      height: 60,
    },
    titleStyle: {},
    disabledStyle: {},
    disabledTitleStyle: {},
    loadingStyle: {},
    iconContainerStyle: {},
  },
  // Button type - outline
  outline: {
    type: 'outline',
    containerStyle: {},
    buttonStyle: {},
    titleStyle: {},
    disabledStyle: {},
    disabledTitleStyle: {},
    loadingStyle: {},
    iconContainerStyle: {},
  },
  // Button type - clear
  clear: {
    type: 'clear',
    containerStyle: {},
    buttonStyle: {},
    titleStyle: {},
    disabledStyle: {},
    disabledTitleStyle: {},
    loadingStyle: {},
    iconContainerStyle: {},
  },
};
