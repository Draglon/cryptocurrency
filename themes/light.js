export default {
  // React Native Elements
  Button: {
    containerStyle: {
      width: 100,
    },
    titleStyle: {
      color: '#fff',
    },
  },
  Icon: {
    type: 'font-awesome',
    size: 24,
  },

  // Custom Elements
  IconClose: {
    containerStyle: {
      backgroundColor: '#ccc',
      color: '#f00',
      position: 'absolute',
      top: 10,
      right: 10,
      width: 30,
      height: 30,
      lineHeight: 30,
      textAlign: 'center',
    },
  },
};
