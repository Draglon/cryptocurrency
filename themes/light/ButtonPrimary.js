// import { COLORS } from '../../constants/variables';

export default {
  // Middle Button - default
  type: 'solid',
  containerStyle: {},
  buttonStyle: {
    paddingLeft: 15,
    paddingRight: 15,
    height: 50,
  },
  titleStyle: {},
  disabledStyle: {},
  disabledTitleStyle: {},
  loadingStyle: {},
  iconContainerStyle: {},
  // Small Button
  small: {
    containerStyle: {},
    buttonStyle: {
      paddingLeft: 10,
      paddingRight: 10,
      height: 30,
    },
    titleStyle: {
      fontSize: 14,
    },
    disabledStyle: {},
    disabledTitleStyle: {},
    loadingStyle: {},
    iconContainerStyle: {},
  },
  // Large Button
  large: {
    containerStyle: {},
    buttonStyle: {
      height: 60,
    },
    titleStyle: {},
    disabledStyle: {},
    disabledTitleStyle: {},
    loadingStyle: {},
    iconContainerStyle: {},
  },
  // Button type - outline
  outline: {
    type: 'outline',
    containerStyle: {},
    buttonStyle: {},
    titleStyle: {},
    disabledStyle: {},
    disabledTitleStyle: {},
    loadingStyle: {},
    iconContainerStyle: {},
  },
  // Button type - clear
  clear: {
    type: 'clear',
    containerStyle: {},
    buttonStyle: {},
    titleStyle: {},
    disabledStyle: {},
    disabledTitleStyle: {},
    loadingStyle: {},
    iconContainerStyle: {},
  },
};
