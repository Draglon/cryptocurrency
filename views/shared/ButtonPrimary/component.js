import React from 'react';
import { StyleSheet, View, Button } from 'react-native';

const ButtonPrimary = () => (
  <View
    style={styles.buttonWrapper}
  >
    <Button
      style={styles.buttonPrimary}
      title="Add to list"
    />
  </View>
);

export default ButtonPrimary;

const styles = StyleSheet.create({
  buttonWrapper: {

  },
  buttonPrimary: {
    backgroundColor: '#ccc',
    flex: 1,
    alignItems: 'stretch',
  },
});
