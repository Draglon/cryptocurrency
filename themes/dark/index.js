import { COLORS } from '../../constants/variables';

import Header from './Header';
import MenuButton from './MenuButton';
import {
  Sidebar, SidebarHeader, SidebarBody, SidebarWrapper,
} from './Sidebar';
import { Menu, MenuItem, MenuItemIcon } from './Menu';
import { Themes, ThemesItem } from './Themes';
import { Languages, LanguagesItem } from './Languages';
import { Page, PageBody, PageWrapper } from './Page';
import { MySets, MySetsEmpty } from './MySets';
import { CreateSet } from './CreateSet';
import Logo from './Logo';
import BackLink from './BackLink';
import ButtonPrimary from './ButtonPrimary';
import ButtonClose from './ButtonClose';

export default {
  // ================= React Native Elements =================
  Divider: {
    backgroundColor: COLORS.white,
    height: 2,
  },
  Text: {
    h1Style: {
      color: COLORS.white,
      marginBottom: 15,
    },
    h2Style: {
      color: COLORS.white,
      marginBottom: 15,
    },
    h3Style: {
      color: COLORS.white,
      marginBottom: 15,
    },
    h4Style: {
      color: COLORS.white,
      marginBottom: 12,
    },
    style: {
      color: COLORS.grayLight,
      marginBottom: 10,
    },
  },
  Icon: {
    type: 'font-awesome',
    size: 18,
  },

  // ================ Custom Elements ========================
  // Header
  Header,
  MenuButton,
  // Pages
  Page,
  PageBody,
  PageWrapper,
  // Sidebar
  Sidebar,
  SidebarHeader,
  SidebarBody,
  SidebarWrapper,
  Menu,
  MenuItem,
  MenuItemIcon,
  Themes,
  ThemesItem,
  Languages,
  LanguagesItem,
  // My sets - page
  MySets,
  MySetsEmpty,
  // Create set - page
  CreateSet,
  // Shared
  Logo,
  BackLink,
  ButtonPrimary,
  ButtonClose,
};
