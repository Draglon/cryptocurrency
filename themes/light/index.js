import { COLORS } from '../../constants/variables';

import Header from './Header';
import MenuButton from './MenuButton';
import {
  Sidebar, SidebarHeader, SidebarBody, SidebarWrapper,
} from './Sidebar';
import { Menu, MenuTitle, MenuItem } from './Menu';
import { Themes, ThemesItem } from './Themes';
import { Languages, LanguagesItem } from './Languages';
import { Page, PageBody, PageWrapper } from './Page';
import { MySets, MySetsEmpty } from './MySets';
import { CreateSet } from './CreateSet';
import Logo from './Logo';
import BackLink from './BackLink';
import ButtonClose from './ButtonClose';
import ButtonPrimary from './ButtonPrimary';
import ButtonCheckbox from './ButtonCheckbox';

export default {
  // ================= React Native Elements =================
  Divider: {
    style: {
      backgroundColor: COLORS.gray,
      height: 2,
    },
  },
  Text: {
    h1Style: {
      color: COLORS.gray,
      marginBottom: 15,
    },
    h2Style: {
      color: COLORS.gray,
      marginBottom: 15,
    },
    h3Style: {
      color: COLORS.gray,
      marginBottom: 15,
    },
    h4Style: {
      color: COLORS.gray,
      marginBottom: 15,
    },
    style: {
      color: COLORS.grayLight,
      marginBottom: 10,
    },
  },
  Icon: {
    type: 'font-awesome',
    size: 18,
  },

  // ================ Custom Elements ========================
  // Header
  Header,
  MenuButton,
  // Pages
  Page,
  PageBody,
  PageWrapper,
  // Sidebar
  Sidebar,
  SidebarHeader,
  SidebarBody,
  SidebarWrapper,
  Menu,
  MenuTitle,
  MenuItem,
  Themes,
  ThemesItem,
  Languages,
  LanguagesItem,
  // My sets - page
  MySets,
  MySetsEmpty,
  // Create set - page
  CreateSet,
  // Shared
  Logo,
  BackLink,
  ButtonClose,
  ButtonPrimary,
  ButtonCheckbox,
};
