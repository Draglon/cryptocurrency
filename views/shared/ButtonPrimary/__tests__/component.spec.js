import React from 'react';
import renderer from 'react-test-renderer';

import { THEMES } from '../../../../themes';
import ButtonPrimary from '../component';

describe('ButtonPrimary component matches snapshot', () => {
  const defaultProps = {
    theme: {
      ButtonPrimary: {
        ...THEMES.light.ButtonPrimary,
      },
    },
    type: '',
    size: '',
    title: '',
    icon: <></>,
    iconRight: false,
    loading: false,
    disabled: false,
    raised: false,
    onPress: jest.fn(),
    customStyle: {},
  };

  it('with default props', () => {
    const wrapper = renderer.create(<ButtonPrimary {...defaultProps} />).toJSON();

    expect(wrapper).toMatchSnapshot();
  });

  it('with dark theme', () => {
    const props = {
      ...defaultProps,
      theme: {
        ButtonPrimary: {
          ...THEMES.dark.ButtonPrimary,
        },
      },
    };
    const wrapper = renderer.create(<ButtonPrimary {...props} />).toJSON();

    expect(wrapper).toMatchSnapshot();
  });

  it('with button type', () => {
    const props = {
      ...defaultProps,
      type: 'outline',
      theme: {
        ButtonPrimary: {
          ...THEMES.dark.ButtonPrimary,
        },
      },
    };
    const wrapper = renderer.create(<ButtonPrimary {...props} />).toJSON();

    expect(wrapper).toMatchSnapshot();
  });

  it('with button size', () => {
    const props = {
      ...defaultProps,
      size: 'small',
      theme: {
        ButtonPrimary: {
          ...THEMES.dark.ButtonPrimary,
        },
      },
    };
    const wrapper = renderer.create(<ButtonPrimary {...props} />).toJSON();

    expect(wrapper).toMatchSnapshot();
  });
});
