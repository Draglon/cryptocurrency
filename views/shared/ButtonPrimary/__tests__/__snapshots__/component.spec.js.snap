// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonPrimary component matches snapshot with button size 1`] = `
<View
  style={
    Object {
      "borderRadius": 3,
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#2089dc",
          "borderColor": "#2089dc",
          "borderRadius": 3,
          "borderWidth": 0,
          "flexDirection": "row",
          "height": 30,
          "justifyContent": "center",
          "padding": 8,
        }
      }
    />
  </View>
</View>
`;

exports[`ButtonPrimary component matches snapshot with button type 1`] = `
<View
  style={
    Object {
      "borderRadius": 3,
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderColor": "#2089dc",
          "borderRadius": 3,
          "borderWidth": 0.5,
          "flexDirection": "row",
          "justifyContent": "center",
          "padding": 8,
        }
      }
    />
  </View>
</View>
`;

exports[`ButtonPrimary component matches snapshot with dark theme 1`] = `
<View
  style={
    Object {
      "borderRadius": 3,
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#2089dc",
          "borderColor": "#2089dc",
          "borderRadius": 3,
          "borderWidth": 0,
          "flexDirection": "row",
          "height": 50,
          "justifyContent": "center",
          "padding": 8,
          "paddingLeft": 15,
          "paddingRight": 15,
        }
      }
    />
  </View>
</View>
`;

exports[`ButtonPrimary component matches snapshot with default props 1`] = `
<View
  style={
    Object {
      "borderRadius": 3,
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#2089dc",
          "borderColor": "#2089dc",
          "borderRadius": 3,
          "borderWidth": 0,
          "flexDirection": "row",
          "height": 50,
          "justifyContent": "center",
          "padding": 8,
          "paddingLeft": 15,
          "paddingRight": 15,
        }
      }
    />
  </View>
</View>
`;
